python_modules/dagit/dagit/webapp/src/App.tsx

```diff
@@ -3,45 +3,80 @@ import gql from "graphql-tag";
 import { Query, QueryResult } from "react-apollo";
 import { Route } from "react-router";
 import { BrowserRouter } from "react-router-dom";
-import Page from "./Page";
 import Loading from "./Loading";
-import Pipelines from "./Pipelines";
+import PipelineExplorer from "./PipelineExplorer";
+import PipelineJumpBar from "./PipelineJumpBar";
 import { AppQuery } from "./types/AppQuery";
 
+import { Alignment, Navbar, NonIdealState } from "@blueprintjs/core";
+import navBarImage from "./images/nav-logo.png";
+
 export default class App extends React.Component {
   public render() {
     return (
-      <BrowserRouter>
-        <Route path="/:pipeline">
-          {({ match, history }) => (
-            <Query query={APP_QUERY}>
-              {(queryResult: QueryResult<AppQuery, any>) => (
-                <Page>
-                  <Loading queryResult={queryResult}>
-                    {data => (
-                      <Pipelines
-                        selectedPipeline={match ? match.params.pipeline : ""}
-                        pipelines={data.pipelines}
-                        history={history}
-                      />
-                    )}
-                  </Loading>
-                </Page>
-              )}
-            </Query>
-          )}
-        </Route>
-      </BrowserRouter>
+      <Query query={APP_QUERY}>
```

Comment:
What I would do is add `navbar` prop to Page and make it accept React.Node, so that you can pass custom header parts to Navbar.